public/class-award-force-sso.php

```diff
@@ -57,14 +58,27 @@ private function getSlug(WP_User $user)
      */
     private function requestSlug(WP_User $user)
     {
-        $response = $this->api->post('/user', [
-            'form_params' => [
-                'email'     => $user->user_email,
-                'firstName' => $user->user_firstname ?: 'First',
-                'lastName'  => $user->user_lastname ?: 'Last'
-            ]
-        ]);
+        try {
+            $response = $this->api->post('/user', [
+                'email' => $user->user_email,
+                'first_name' => $user->user_firstname ?: 'First',
+                'last_name' => $user->user_lastname ?: 'Last',
+                'password' => uniqid(),
+            ]);
 
+            if (!$response) {
+                return $this->requestSlugByEmail($user->user_email);
+            }
+
+            return $response->slug;
+        } catch (Exception $e) {
+            return $this->requestSlugByEmail($user->user_email);
+        }
```

Comment:
I'm still a bit confused by this. It's not very clear how `post` works in the client.
- When does it throw an exception, and why is it handled out here and not in the client? I can't think of any exception where we would want to continue calling the API.
- When does the post not return a response? Would it make sense to try `$this->requestSlugByEmail($user->user_email)` first, and if that does not return a response then we post?

Reply:
OK, your last comment made me think about the behavior of the plugin. I've changed a bit the way it works. Now:
- First tries to grab the slug using the email
- If the user exists, grabs the slug from the response.
- If the user doesn't  exist, **only in this case** tries to create the user on AF/GG and grabs the slug from the response (`Location` header)